visualization/app/codeCharta/util/gameObjectsParser/gameObjectsImporter.ts

```diff
@@ -0,0 +1,137 @@
+import { ExportWrappedCCFile } from "../../codeCharta.api.model"
+
+export function parseGameObjectsFile(data) {
+	const { gameObjectPositions, cycles = [] } = JSON.parse(data)
+
+	const ccJson: ExportWrappedCCFile = {
+		checksum: "",
+		data: {
+			projectName: "ScriptProjectName",
+			fileChecksum: "",
+			apiVersion: "1.3",
+			nodes: []
+		}
+	}
+
+	const nodes = [{ name: "base", type: NodeType.FOLDER, attributes: {}, children: [] }]
+	gameObjectPositions.push(createBaseGameObjectPosition(gameObjectPositions))
+
+	for (const gameObjectPosition of gameObjectPositions) {
+		const nodeNames = gameObjectPosition.name.split(".")
+		if (nodeNames[0] !== "base") {
+			addNodeRecursively(nodeNames, nodes[0].children, "base", gameObjectPosition, gameObjectPositions)
+		}
+	}
+
+	ccJson.data.nodes = nodes
+	ccJson.data.edges = cycles.map(cycle => createEdge(cycle))
+	ccJson.data.attributeTypes = createAttributeTypes()
+	ccJson.checksum = md5(JSON.stringify(ccJson.data))
+	return ccJson
+}
+
+function addNodeRecursively(nodeNames: string[], nodes: CodeMapNode[], parentNodeName: string, gameObjectPosition, gameObjectPositions) {
+	if (nodeNames.length === 0) return
+
+	// get current node name and create cc node structure
+	const [nodeName] = nodeNames
+	let node: CodeMapNode = {
+		name: nodeName,
+		type: isFile(nodeNames) ? NodeType.FILE : NodeType.FOLDER,
+		attributes: {}
+	}
+	if (!isFile(nodeNames)) node.children = []
+	if (isFile(nodeNames)) node = wrapFileInAFolder(nodeName, node, gameObjectPosition)
+
+	// add or select node if found
+	if (!nodeAlreadyExists(nodes, nodeName)) {
+		nodes.push(node)
+	} else {
+		node = nodes.find(node => node.name === nodeName)
+	}
+
+	const parent = gameObjectPositions.find(gameObject => gameObject.name === parentNodeName)
+	node.fixedPosition = calculateFixedFolderPosition(node, nodeName, parent, gameObjectPosition)
+
+	const newParentName = parentNodeName === "base" ? node.name : `${parentNodeName}.${node.name}`
+	addNodeRecursively(nodeNames.slice(1), node.children, newParentName, gameObjectPosition, gameObjectPositions)
+}
+
+function isFile(names: string[]) {
+	return names.length === 1
+}
+
+function nodeAlreadyExists(nodes: CodeMapNode[], name: string) {
+	return nodes.some(node => node.name === name)
+}
+
+function wrapFileInAFolder(nodeName: string, node: CodeMapNode, gameObjectPosition) {
+	const childNode = { ...node }
+	childNode.attributes = { height: gameObjectPosition.scale.y }
+	const parentNode: CodeMapNode = { name: nodeName, type: NodeType.FOLDER, attributes: {}, children: [childNode] }
+	return parentNode
+}
+
+function calculateFixedFolderPosition(node: CodeMapNode, name: string, parentGameObject, childGameObject) {
+	let position: FixedPosition
+	if (node.type === NodeType.FOLDER && name !== "base") {
+		// translate the center point from the middle of the gameObject to the corner as needed for fixedPosition
```

Comment:
I'm not sure if it explains why we calculate top, bottom, width and height

Reply:
Take a look at our fixed folders' documentation, it will explain what we are doing and why 👍 
https://maibornwolff.github.io/codecharta/how-to/fixate_folders_with_a_custom_cc_json/